appserver/payara-appserver-modules/microprofile/fault-tolerance/src/main/java/fish/payara/microprofile/faulttolerance/cdi/FaultToleranceInterceptor.java

```diff
@@ -74,9 +77,12 @@ public Object intercept(InvocationContext context) throws Exception {
         try {
             FaultToleranceService env =
                     Globals.getDefaultBaseServiceLocator().getService(FaultToleranceService.class);
-            FaultTolerancePolicy policy = FaultTolerancePolicy.get(context, () -> env.getConfig(context, this));
+            AtomicReference<FaultToleranceConfig> lazyConfig = new AtomicReference<>();
```

Comment:
NB. Makes sure multiple calls to the supplier function always result in the same config instance.